tests/test_urlhaus.py

```diff
@@ -0,0 +1,469 @@
+from unittest.mock import patch, Mock
+import pytest
+import requests
+import json
+from modules.threat_intelligence.urlhaus import (
+    URLhaus,
+    ThreatLevel,
+    EvidenceType,
+    IDEACategory,
+    Direction,
+    URLHAUS_BASE_URL,
+)
+
+
+@pytest.mark.parametrize(
+    "to_lookup, expected_url, status_code",
+    [
+        # Testcase1: URL lookup
+        (
+            {"url": "https://example.com"},
+            f"{URLHAUS_BASE_URL}/url/",
+            200,
+        ),
+        # Testcase2: hash lookup
+        (
+            {"md5_hash": "a1b2c3d4"},
+            f"{URLHAUS_BASE_URL}/payload/",
+            200,
+        ),
+    ],
+)
+@patch("modules.threat_intelligence.urlhaus.requests.session")
+def test_make_urlhaus_request(
+    mock_response, mock_db, to_lookup, expected_url, status_code
+):
+    """Test successful requests to the make_urlhaus_request function."""
+    mock_response_instance = Mock()
+    mock_response.return_value = mock_response_instance
+    mock_response = Mock(status_code=status_code)
+    mock_response_instance.post.return_value = mock_response
+
+    urlhaus = URLhaus(mock_db)
```

Comment:
this should be moved to module_factory, always remember no object ever used in the tests should be created in a test_*.py file, because you'll need to recreate it again later and that would cause code duplication

Reply:
check any object creation in all functions for the same issue ok?

Reply:
Sorry i thought you were only asking about only the object which were being used as fixture in the code 
i am fixing it.